src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java

```diff
@@ -1,48 +1,94 @@
 
-    public LogCacheClient(ConnectionContext connectionContext, String logCacheApi, TokenProvider tokenProvider,
-                          Map<String, String> requestTags) {
-        super(connectionContext, Mono.just(logCacheApi), tokenProvider, requestTags);
-        this.mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-                                        .configure(DeserializationFeature.UNWRAP_ROOT_VALUE, true);
+    public LogCacheClient(String logCacheApi, OAuthClient oAuthClient, Map<String, String> requestTags) {
+        this.client = HttpClient.newBuilder()
+                                .followRedirects(HttpClient.Redirect.NORMAL)
+                                .connectTimeout(Duration.ofMinutes(30))
+                                .build();
+        this.logCacheApi = logCacheApi;
+        this.oAuthClient = oAuthClient;
+        this.requestTags = requestTags;
     }
 
-    public Flux<ApplicationLogEntity> getRecentLogs(UUID applicationGuid) {
-        return createOperator().flatMapMany(operator -> operator.get()
-                                                                .uri(builder -> getEndpoint(builder, applicationGuid.toString()))
-                                                                .response()
-                                                                .parseBodyToMono(this::parseBody)
-                                                                .flatMapIterable(ApplicationLogsResponse::getLogs));
+    public Flux<ApplicationLogEntity> getRecentLogs(UUID applicationGuid, LocalDateTime offset) {
+        HttpRequest request = buildRequest(applicationGuid, offset);
+
+        HttpResponse<InputStream> response;
+        try {
+            response = client.send(request, BodyHandlers.ofInputStream());
+        } catch (IOException | InterruptedException e) {
+            throw new CloudException(e.getMessage(), e);
+        }
+
+        if (response.statusCode() / 100 != 2) {
+            var status = HttpStatus.valueOf(response.statusCode());
+            throw new CloudOperationException(status, status.getReasonPhrase(), parseBodyToString(response.body()));
+        }
+        return Flux.fromIterable(parseBody(response.body()).getLogs());
+    }
+
+    private HttpRequest buildRequest(UUID applicationGuid, LocalDateTime offset) {
+        var instant = offset.toInstant(ZoneOffset.UTC);
+        var secondsInNanos = Duration.ofSeconds(instant.getEpochSecond())
+                                                       .toNanos();
+        var requestBuilder = HttpRequest.newBuilder()
+                                        .GET()
+                                        .uri(URI.create(logCacheApi + "/api/v1/read/" + applicationGuid
+                                                + "?envelope_types=LOG&descending=true&limit=1000&start_time="
+                                                + (secondsInNanos + instant.getNano() + 1)))
```

Comment:
Use String.format or StringBuilder instead of concatenating strings because it is a bit hard to read this way